src/buffer/out/TextColor.cpp

```diff
@@ -156,13 +156,67 @@ COLORREF TextColor::GetColor(gsl::span<const COLORREF> colorTable,
             //      (Settings::_DefaultForeground==INVALID_COLOR, and the index
             //      from _wFillAttribute is being used instead.)
             // If we find a match, return instead the bright version of this color
+
+            static_assert(sizeof(COLORREF) * 8 == 32, "The vectorized code broke. If you can't fix COLORREF, just remove the vectorized code.");
+
+#pragma warning(push)
+#pragma warning(disable : 26481) // Don't use pointer arithmetic. Use span instead (bounds.1).
+#pragma warning(disable : 26490) // Don't use reinterpret_cast (type.1).
+#ifdef __AVX2__
+            // I wrote this vectorized code one day, because the sun was shining so nicely.
+            // There's no other reason for this to exist here, except for being pretty.
+            // This code implements the exact same for loop you can find below, but is ~3x faster.
+            //
+            // A brief explanation for people unfamiliar with vectorized instructions:
+            //   Vectorized instructions, like "SSE" or "AVX", allow you to run
+            //   common operations like additions, multiplications, comparisons,
+            //   or bitwise operations concurrently on multiple values at once.
+            //
+            // We want to find the given defaultColor in the first 8 values of colorTable.
+            // Coincidentally a COLORREF is a DWORD and 8 of them are exactly 256 bits.
+            // -- The size of a single AVX register.
+            //
+            // Thus, the code works like this:
+            // 1. Load all 8 DWORDs at once into one register
+            // 2. Set the same defaultColor 8 times in another register
+            // 3. Compare all 8 values at once
+            //    The result is either 0xffffffff or 0x00000000.
+            // 4. Extract the most significant bit of each DWORD
+            //    Assuming that no duplicate colors exist in colorTable,
+            //    the result will be something like 0b00100000.
+            // 5. Use BitScanForward (bsf) to find the index of the most significant 1 bit.
+            const auto haystack = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(colorTable.data())); // 1.
+            const auto needle = _mm256_set1_epi32(__builtin_bit_cast(int, defaultColor)); // 2.
+            const auto result = _mm256_cmpeq_epi32(haystack, needle); // 3.
+            const auto mask = _mm256_movemask_ps(_mm256_castsi256_ps(result)); // 4.
+            unsigned long index;

```

Comment:
Does this need to get initialized to 0?

Reply:
`_BitScanForward` compiles to the assembly instruction `bsf`, whose resulting value (the one stored in `index`) is undefined if the given value has no set bits. Since I only use `index` when `_BitScanForward` return true (meaning at least 1 bit was set) I'm never reading any uninitialized values.

Reply:
okeydokey good enough for me!